modules/ximgproc/src/deriche_filter.cpp

```diff
@@ -0,0 +1,410 @@
+#include "precomp.hpp"
+#include "opencv2/highgui.hpp"
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+namespace cv {
+namespace ximgproc {
+    template<typename T>
+    static void VerticalIIRFilter(Mat img,Mat dst,const Range r,T *c1,double alphaDerive)
```

Comment:
I agree, it will be better to make this variable local and call the function with explicit template specification: `VerticalIIRFilter<uchar>(...`.